### PR TITLE
SONAR-9356 new webservice stub for api/users/skipOnboardingTutorial

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/SkipOnboardingTutorialAction.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+
+public class SkipOnboardingTutorialAction implements UsersWsAction {
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction("skipOnboardingTutorial")
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/SkipOnboardingTutorialAction.java]: MAJOR Remove this useless assignment to local variable "action". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1854)
MINOR Remove this unused "action" local variable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1481)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: it/it-tests/src/test/java/it/user/SkipOnboardingTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  @ClassRule
+  public static final Orchestrator orchestrator = Category4Suite.ORCHESTRATOR;
+  @ClassRule
+  public static UserRule userRule = UserRule.from(orchestrator);
+
+  @Test
+  public void should_operate_silently() {
+    String login = randomAlphabetic(10).toLowerCase();
+    String password = randomAlphabetic(10).toLowerCase();
+    userRule.createUser(login, password);
```
Comment (julienlancelot) [inline it/it-tests/src/test/java/it/user/SkipOnboardingTest.java]: Dont' forget to remove the user at the end of the test

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/SkipOnboardingTutorialAction.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+public class SkipOnboardingTutorialAction implements UsersWsAction {
+
+  @Override
+  public void define(WebService.NewController context) {
+    context.createAction("skipOnboardingTutorial")
+      .setPost(true)
+      .setInternal(true)
+      .setDescription("Stores that the user has skipped the onboarding tutorial and does not want to see it after future logins." +
+        " Calling this webservice several times will silently ignore subsequent requests.")
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/SkipOnboardingTutorialAction.java]: You can also say that this WS needs the user to be authenticate (as it should fail if this web service is called anonymously)

File: server/sonar-server/src/test/java/org/sonar/server/user/ws/SkipOnboardingTutorialActionTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+public class SkipOnboardingTutorialActionTest {
+
+  @Test
+  public void should_have_a_good_definition() {
+    WsActionTester ws = new WsActionTester(new SkipOnboardingTutorialAction());
+    WebService.Action def = ws.getDef();
+    assertThat(def.isPost()).isTrue();
+    assertThat(def.isInternal()).isTrue();
+    assertThat(def.since()).isEqualTo("6.5");
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/user/ws/SkipOnboardingTutorialActionTest.java]: Add a test that there's no parameter

File: it/it-tests/src/test/java/it/user/SkipOnboardingTest.java (hunk 2 above)
Comment (bartfastiel) [inline it/it-tests/src/test/java/it/user/SkipOnboardingTest.java]: What do you think about this:
```
    } finally {
      userRule.deactivateUsers(login);
    }
```

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/SkipOnboardingTutorialAction.java (hunk 3 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/SkipOnboardingTutorialAction.java]: Currently it works fine, if the user is not authenticated. I'd like to fix this in the ticket about the implementation of that ws. (Maybe this is even a good entry point for tdd ;) )

File: server/sonar-server/src/test/java/org/sonar/server/user/ws/SkipOnboardingTutorialActionTest.java (hunk 4 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/user/ws/SkipOnboardingTutorialActionTest.java]: I added it, but I would love to discuss it in our next coffee break ;)

**Pull request merged**